main.cc

```diff
@@ -1173,6 +1174,15 @@ sharded<locator::shared_token_metadata> token_metadata;
             auto stop_dsm = defer_verbose_shutdown("disk space monitor", [&disk_space_monitor_shard0] {
                 disk_space_monitor_shard0->stop().get();
             });
+            auto out_of_space_subscription = disk_space_monitor_shard0->subscribe(cfg->critical_disk_utilization_level, [&threhsold = cfg->critical_disk_utilization_level, &dsm = *disk_space_monitor_shard0] (auto threshold_reached) {
+                static constexpr auto msg_template = "{} the critical disk utilization level ({:.1f}%). Current disk utilization {:.1f}%";
+                if (bool(threshold_reached)) {
```

Comment:
nit: is the explicit cast to bool needed?

Reply:
Yes, because of the `explicit operator bool()`  in the `bool_cass`

Reply:
It should not be needed, see Contextual Conversions in https://en.cppreference.com/w/cpp/language/implicit_conversion.html.

Reply:
Correct. Fixed.